### ISSUE TITLE
Amd ADL library not found

### ISSUE THREAD

**Issue by riodda** [nanopool/Claymore-Dual-Miner#139]
Would be great to have a couple of tips in how to install the AMD ADL library

**Comment by jojoCoWorker** [comment]
I'm looking for answers too.... ;-)

**Comment by ZzAntares** [comment]
Hi @riodda did you managed to solve it? I'm experiencing the same problem here 😩

**Comment by zmcartor** [comment]
Also see same message. Launched w `-asm 0` option , single mining ETH. Any help appreciated!

**Comment by hasJoshT** [comment]
Getting the same message. As far as I know, this means that you are using a video card driver that doesn't include the AMD ADL library. I'm on Linux using the AMDGPU-PRO Beta Mining Driver 17.40, and that doesn't include ADL. After doing some searching, it appears that this means mining will function correctly, but Claymore can't control your settings (like overclocking) for you. 

My next step here is to uninstall the AMDGPU-PRO beta driver and install the standard AMDGPU driver which _does_ include ADL. That should give Claymore what it needs to manage overclocking.

Ideally, if Claymore would update and patch in the ADL for AMDGPU-PRO drivers, that would be the best solution for the future.

**Comment by vtodo** [comment]
Is there a workaround this?

**Comment by alichaudry** [comment]
Still an issue for me. Any ideas?

**Comment by alxcancado** [comment]
Me too! What to do? I'm on Linux, ethOS (Ubuntu)

**Comment by Ben-and-Ellen** [comment]
Tried AMD drivers 17.40 and 17.50, both report that 
"AMD ADL library not found"

**Comment by chiefddj** [comment]
any solution ?

**Comment by stevebills** [comment]
dont over clock may sort it

**Comment by D3m0nKingx** [comment]
I'm looking for the ADL files as well, I just found today AMGPU-PRO 18.10 : https://support.amd.com/en-us/kb-articles/Pages/Radeon-Software-for-Linux-Release-Notes.aspx

Anybody tested it?

**Comment by D3m0nKingx** [comment]
Looks like they included the header files (from AMD_ADL). just have to do `sudo apt  -f install opencl-amdgpu-{pro,pro-dev}`
after you've installed the amdgpu-pro package: `sudo ./amdgpu-pro-install`

Afterwards, add this line for the *configure* script* to point to the necessary files:

`CFLAGS="-O2 -Wall -march=native-I /opt/amdgpu-pro/include/" LDFLAGS="-L/opt/amdgpu/lib/x86_64-linux-gnu" ./configure`

**Comment by D3m0nKingx** [comment]
I placed the ADL files in a zip for ppl that need it:

https://mega.nz/#!8x4B2Ira!kOyy9ZrcvYAIaSp4iIknXxUoDnuqIBnp64KRxyndC9E

Included files:

adl_defines.h  adl_sdk.h  adl_structures.h  readme.txt

**Comment by trace86** [comment]
Hey DemonKingx

I've followed your guide but dunno where the configure script is, can you point me to the right direction?

**Comment by D3m0nKingx** [comment]
Hi @trace86, could you be more specific the steps you took. thanks

**Comment by MoneroCrusher** [comment]
@D3m0nKingx I think this is a general problem with the 17.40 driver. in which path are we supposed to put these ADL files?

**Comment by D3m0nKingx** [comment]
The ADL files are for when actually compiling the miner software. 

Such as when compiling cgminer you would place the files in `cgminer-3.7.2-kalroth/ADL_SDK` folder then run  
`CFLAGS="-O2 -Wall -march=native -I /opt/amdgpu-pro/include/" LDFLAGS="-L/opt/amdgpu-pro/lib/x86_64-linux-gnu" ./configure` command within cgminer-3.7.2-kalroth folder. 

**AFTER** having installed the amdgpu pro drivers (https://support.amd.com/en-us/kb-articles/Pages/Radeon-Software-for-Linux-Release-Notes.aspx) Latest release is now v18.20 driver support for Ubuntu 16 and 18.

However with Claymore-Dual-Miner I don't actually even see a downloadable source code to actually compile the software locally ??? @MoneroCrusher

**Comment by trace86** [comment]
@D3m0nKingx I was trying to run this code but returned this

CFLAGS="-O2 -Wall -march=native -I /opt/amdgpu-pro/include/" LDFLAGS="-L/opt/amdgpu-pro/lib/x86_64-linux-gnu" ./configure
-bash: ./configure: No such file or directory

I've used apt-install to insall the headers (Looks like they included the header files (from AMD_SDK). just have to do sudo apt -f install opencl-amdgpu-{pro,pro-dev}) and I'm using Claymore currently.  That's why i'm wondering where the configuration files would be installed

**Comment by D3m0nKingx** [comment]
@trace86 what folder are you running the code from ?

**Comment by D3m0nKingx** [comment]
Please note the AMD  ADL library is not part of the amdgpu drivers package, this is a separate library ! Which I've uploaded https://mega.nz/#!8x4B2Ira!kOyy9ZrcvYAIaSp4iIknXxUoDnuqIBnp64KRxyndC9E and you can also get from sourceforge (https://sourceforge.net/projects/nicehashsgminerv5viptools/files/AMD%20Display%20Library%20%28ADL%29%20SDK/) v10.2

However, could you please provide me the exact error output you're getting from running the miner ? is this on windows or linux ? @trace86

**Comment by trace86** [comment]
@D3m0nKingx This is linux.  I must've misunderstood then, this means I need to download the library into the miners folder, and then unzip it, and then execute `sudo apt -f install opencl-amdgpu-{pro,pro-dev}` ?

**Comment by D3m0nKingx** [comment]
> This is linux. I must've misunderstood then, this means I need to download the library into the miners folder, and then unzip it, and then execute sudo apt -f install opencl-amdgpu-{pro,pro-dev} ?

Right the library goes into the miner's folder. Can you provide me a link to the actual source code to compile that you're using ?

For the amdgpu-pro drivers installation you just need to do `sudo ./amdgpu-pro-install --headless --opencl=legacy,pal` from within the amdgpu-pro folder you downloaded/extracted.

**Comment by trace86** [comment]
i've used the amdgpu-pro drivers from the official website here
[](https://support.amd.com/en-us/kb-articles/Pages/Radeon-Software-for-Linux-Release-Notes.aspx)

`sudo ./amdgpu-pro-install --opencl=pal`

and then i've download the claymore from here
[](https://github.com/nanopool/Claymore-Dual-Miner/releases)

That's basically it

**Comment by trace86** [comment]
Yup, still giving me the same error, i've already placed the folder inside claymore forlder as `claymore/ADL_SDK`

~/mining/claymore$ `
CFLAGS="-O2 -Wall -march=native -I /opt/amdgpu-pro/include/" LDFLAGS="-L/opt/amdgpu-pro/lib/x86_64-linux-gnu" ./configure`
-bash: ./configure: No such file or directory

I don't see a configure file to run though within the miner

**Comment by D3m0nKingx** [comment]
@trace86 @MoneroCrusher Ok so I did a strace on the miner and this is the output:

```bash
open("/lib/x86_64-linux-gnu/libatiadlxx.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/usr/lib/x86_64-linux-gnu/libatiadlxx.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/lib/libatiadlxx.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
open("/usr/lib/libatiadlxx.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
munmap(0x7f6567dda000, 146284)          = 0
write(1, "\33[0mAMD ADL library not found.\n", 31) = 31
```
It's not what we thought lol, it's different set of files it's searching, but good news it's irrelevant as those files are only necessary specifically to Linux Fedora Distribution. (see, https://www.rpmfind.net/linux/RPM/rpmfusion/nonfree/fedora/updates/21/i386/xorg-x11-drv-catalyst-libs-15.9-1.fc21.i686.html ) 

Issue is resolved ! The miner is operating just as it should, the error can be ignored for Ubuntu :)
